src/color.rs

```diff
@@ -78,120 +96,97 @@ impl Color {
             Self::BrightMagenta => "105".into(),
             Self::BrightCyan => "106".into(),
             Self::BrightWhite => "107".into(),
-            Self::TrueColor { .. } if !truecolor_support() => {
-                self.closest_color_euclidean().to_bg_str()
+            Self::Ansi256 { idx } => match get_current_color_level() {
+                ColorLevel::Ansi16 => self.ansi256_fallback_to_ansi16().to_bg_str(),
+                ColorLevel::Ansi256 | ColorLevel::TrueColor | ColorLevel::None => {
+                    format!("48;5;{idx}").into()
+                }
+            },
+            Self::TrueColor { r, g, b } => match get_current_color_level() {
+                ColorLevel::Ansi16 => self.truecolor_fallback_to_ansi16().to_bg_str(),
+                ColorLevel::Ansi256 => self.truecolor_fallback_to_ansi256().to_bg_str(),
+                ColorLevel::TrueColor | ColorLevel::None => format!("48;2;{r};{g};{b}").into(),
+            },
+        }
+    }
+
+    /// Converts an ANSI 256-color to the closest ANSI 16-color palette color.
+    #[must_use]
+    pub fn ansi256_fallback_to_ansi16(self) -> Self {
+        match self {
+            Self::Ansi256 { idx } => {
+                let (r, g, b) = ansi256_to_rgb(idx);
+                let mut min_distance_sq = u32::MAX;
+                let mut closest_color = self;
+
+                for &(cr, cg, cb, color) in &ANSI_16_COLORS {
+                    let dr = (i32::from(r) - i32::from(cr)).pow(2) as u32;
+                    let dg = (i32::from(g) - i32::from(cg)).pow(2) as u32;
+                    let db = (i32::from(b) - i32::from(cb)).pow(2) as u32;
+                    let distance_sq = dr + dg + db;
+
+                    if distance_sq < min_distance_sq {
+                        min_distance_sq = distance_sq;
+                        closest_color = color;
+                    }
+                }
+
+                closest_color
             }
-            Self::TrueColor { r, g, b } => format!("48;2;{r};{g};{b}").into(),
+            _ => self,
```

Comment:
I believe functions like these should be kept private.

The fallback should panic: `_ => unreachable!()` (or `unimplemented!()`), because it simply *should never* be called without the appropriate variant.

When the functions are private, we can let them panic, because we can trust that users won't use them incorrectly.